instances/widgets.treasury-factory.near/widget/pages/settings/system-updates/AvailableUpdates.jsx

```diff
@@ -42,6 +50,76 @@ const Container = styled.div`
   }
 `;
 
+async function checkForWidgetUpdate() {
+  const BOOTSTRAP_WIDGET_ACCOUNT = "bootstrap.treasury-factory.near";
+  console.log("checking for widget update");
+  asyncFetch(`${REPL_RPC_URL}`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      jsonrpc: "2.0",
+      id: "dontcare",
+      method: "query",
+      params: {
+        request_type: "call_function",
+        finality: "final",
+        account_id: "social.near",
+        method_name: "get",
+        args_base64: btoa(
+          JSON.stringify({
+            keys: [`${BOOTSTRAP_WIDGET_ACCOUNT}/widget/app`],
+          })
+        ),
+      },
+    }),
+  }).then((response) => {
+    const bootstrap_widget_content = Buffer.from(
+      new Uint8Array(response.body.result.result)
+    ).toString("utf-8");
+
+    asyncFetch(`${REPL_RPC_URL}`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "dontcare",
+        method: "query",
+        params: {
+          request_type: "call_function",
+          finality: "final",
+          account_id: "social.near",
+          method_name: "get",
+          args_base64: btoa(
+            JSON.stringify({
+              keys: [`${instance}/widget/app`],
+            })
+          ),
+        },
+      }),
+    }).then((response) => {
+      const instance_widget_content = Buffer.from(
+        new Uint8Array(response.body.result.result)
+      ).toString("utf-8");
+      if (
+        instance_widget_content ===
+        bootstrap_widget_content.replaceAll(BOOTSTRAP_WIDGET_ACCOUNT, instance)
+      ) {
```

Comment:
I just saw, the app.jsx is different in our older instances, we will need to update those, otherwise they will see the review banner, when you run locally do you see it?
<img width="1479" alt="image" src="https://github.com/user-attachments/assets/4374cc08-1190-49fd-896e-82c18d6bdf3b" />

Reply:
merging this will not show the review banner, since there are no updates published in [UpdateRegistry.jsx](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/staging/instances/widgets.treasury-factory.near/widget/pages/settings/system-updates/UpdateRegistry.jsx).

So the banner and updates in available updates will only show when we publish to UpdateRegistry

Reply:
oh yes that's right, it's fine then